### PR TITLE
dnn: fix some tests

### PR DESCRIPTION
**Pull request by mshabunin**
* Extracted similar code to a function
* Check result of `map::find`

<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac
build_image:Custom=ubuntu-openvino-2020.4.0:16.04
build_image:Custom Win=openvino-2020.4.0
build_image:Custom Mac=openvino-2020.4.0

test_modules:Custom=dnn,python2,python3,java
test_modules:Custom Win=dnn,python2,python3,java
test_modules:Custom Mac=dnn,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/dnn/test/test_ie_models.cpp [hunk 1]
```diff
@@ -108,6 +108,10 @@ static const std::map<std::string, OpenVINOModelTestCaseInfo>& getOpenVINOTestMo
             "intel/person-detection-retail-0013/FP32/person-detection-retail-0013",
             "intel/person-detection-retail-0013/FP16/person-detection-retail-0013"
         }},
+        { "age-gender-recognition-retail-0013", {
```
Comment (alalek) [inline modules/dnn/test/test_ie_models.cpp:111]: Please update the "how to" comment above

File: modules/dnn/test/test_ie_models.cpp [hunk 2]
```diff
@@ -276,6 +280,21 @@ void runCV(Backend backendId, Target targetId, const std::string& xmlPath, const
...
+    const auto it = models.find(modelName);
+    if (it != models.end())
+    {
+        OpenVINOModelTestCaseInfo modelInfo = it->second;
+        if (isFP16 && modelInfo.modelPathFP16)
+            return std::string(modelInfo.modelPathFP16);
+        else if (!isFP16 && modelInfo.modelPathFP32)
+            return std::string(modelInfo.modelPathFP32);
+    }
+    return std::string();
```
Comment (alalek) [inline modules/dnn/test/test_ie_models.cpp:295]: IMHO, error with description or skip action should be raised.
Test code should not allow to hide mistakes.

Comment (mshabunin) [reply modules/dnn/test/test_ie_models.cpp]: I wanted this routine to be separate from testing framework to allow calling it from some static initializer or test fixture constructors. Problems should be caught later by the test body in `findDataFile` and `string::empty()` checks.

File: modules/dnn/test/test_ie_models.cpp (hunk 1 above)
Comment (mshabunin) [reply modules/dnn/test/test_ie_models.cpp:111]: Done.

**Review by alalek — APPROVED** [review]
Thank you!

**Pull request merged**